src/main/java/frc/robot/subsystems/Indexer.java

```diff
@@ -86,103 +91,91 @@ public CommandBase reverseOutput() {
     // This command will make sure that the singulator has possesion of the ball
 
     public CommandBase pierrePossesion() {
-        return new FunctionalCommand(() -> {
-
+        CommandBase cmd = new FunctionalCommand(() -> {
         }, () -> {
-
             if (frontIntakeIR.getAsBoolean()) {
-
                 singulator.set(singulatorMotorSpeed);
             }
-            if (backIntakeIR.getAsBoolean()) {
+            if (backIntakeIR.getAsBoolean() && !topPierreIR.getAsBoolean()) {
```

Comment:
We should add comments explaining why we did this since it's non-obvious.
Also reading the docs `isFinished` gets called IMMEDIATELY after `execute` so I have no idea why the motors have time to run.

Reply:
FIxed